_source/_posts/2021-09-30-oauth-sessions-with-java.adoc

```diff
@@ -0,0 +1,316 @@
+---
+layout: blog_post
+title: "Session Clustering for OAuth 2.0 Applications"
+author: brian-demers
+by: advocate
+communities: [java]
+description: "Learn how sessions are used with OAuth 2.0 and build an example with HAproxy, Redis, and Spring Boot."
+tags:  [spring-session, spring, microservices, jhipster, java, spring-boot, haproxy, oauth2, oidc, session]
+tweets:
+- "Apps using OIDC/OAuth 2.0 don't have 'special' sessions requirements, but you still need to think about them 🧠"
+- "Confused how #OAuth2 and affects your web app's sessions? Learn more and build an example with #Redis, #HAproxy, and #SpringBoot 💡"
+- "Scaling an app from 1 to N, means you need to think about load balancing, session management, and how those relate to Auth🔒"
+image:
+type: conversion
+---
+
+:toc: macro
+:page-liquid:
+:experimental:
+
+A common OAuth 2.0 question we hear is: "How do I deal with OAuth with a load-balanced application". The short answer to the question is there is nothing specific about session clustering for OAuth. The longer answer is, you likely still need to worry about cluster session management. This post will discuss how an OAuth login relates to your application's session and build a simple, secure, load-balanced application.
```

Comment:
```suggestion
A common OAuth 2.0 question we get: "How do I deal with OAuth in a load-balanced application?" The short answer: There's nothing specific about session clustering for OAuth. The longer answer is—you likely still need to worry about cluster session management. This post will discuss how an OAuth login relates to your application's session. And we'll build a simple, secure, load-balanced application to illustrate.
```
smoothing the flow and removing some repetitions... also, in US English, the quotes go outside the other punctuation, this is not the same in UK English.

Reply:
Typo on the quote punctuation, but I was just reading about that, I didn't realize that was a difference between UK and US English!